src/MauiReactor/Internals/ServiceCollectionProvider.cs

```diff
@@ -135,7 +142,7 @@ public sealed class ServiceContext : IDisposable
 {
     public ServiceContext(IServiceProvider serviceProvider)
     {
-        ServiceCollectionProvider.ServiceProvider = serviceProvider;
+        ServiceCollectionProvider.SetTestingServiceProvider(serviceProvider);

```

Comment:
if this context is only used in tests, this should not be a breaking change. If someone is using this class to set the service provider inside of a running application, this will cause unintended side effects. I think since the intent of the class is for testing, this would not be a breaking change, but the behavior is quite a bit different under the hood.